srml/contract/src/wasm/prepare.rs

```diff
@@ -229,7 +229,10 @@ impl<'a, Gas: 'a + As<u32> + Clone> ContractModule<'a, Gas> {
 				.get(*type_idx as usize)
 				.ok_or_else(|| "validation: import entry points to a non-existent type")?;
 
-			if !C::can_satisfy(import.field().as_bytes(), func_ty) {
+			// We disallow importing `gas` function here since it is treated as implementation detail.
+			if import.field().as_bytes() == b"gas"
```

Comment:
Will we have multiple functions that are not allowed? If yes, maybe put `gas` into a static array?

Reply:
I don't expect other functions like that. I'd prefer to keep it simple for now, but yeah whenever there are more functions happy to go with array.